src/openlrr/game/front/Loader.cpp

```diff
@@ -22,18 +26,241 @@ LegoRR::Loader_Globs & LegoRR::loaderGlobs = *(LegoRR::Loader_Globs*)0x00556e00;
+			}
+			Gods98::Mem_Free((void*)localLoaderSec->name);
+		}
+		//Bump up iterator unitl done
+		localLoaderSec = localLoaderSec + 1;
+	} while (localLoaderSec < (LoaderSection*)&loaderGlobs.current);
+
+	//if the file isnt null we're done
+	if (f != nullptr) {
+		Gods98::File_Close(f);
+	}
+	//Remove and close load/shutdown screens
+	Gods98::Image_Remove(loaderGlobs.LoadScreen);
+	if (loaderGlobs.ShutdownScreen != nullptr) {
+		Gods98::Image_Remove(loaderGlobs.ShutdownScreen);
+	}
+	return;
+}
 
 // <LegoRR.exe @0044e0a0>
-//void __cdecl LegoRR::Loader_display_loading_bar(const char* section);
+void __cdecl LegoRR::Loader_display_loading_bar(const char* section)
+{
+	LoaderSection* localLoaderSec;
+	int secCntr;
+	//If section is null clear all flags and bails
+	if (section == nullptr) {
+		loaderGlobs.flags = loaderGlobs.flags & (~LOADER_GLOB_FLAG_ENABLED);
+		return;
+	}
+
+	secCntr = 0; 
+	localLoaderSec = loaderGlobs.sectionList;
+	do {
+		//Check for empty section to determine if we're done
+		if ((localLoaderSec->name != nullptr) &&  (_stricmp(localLoaderSec->name, section) == 0)) {
+			loaderGlobs.current = loaderGlobs.sectionList + secCntr;
+			loaderGlobs.sectionList[secCntr].currentSize = 0;
+			break;
+		}
+		localLoaderSec = localLoaderSec + 1;
+		secCntr = secCntr + 1;
+	//Leave if we pass the end of the current loader struct
+	} while (localLoaderSec < (LoaderSection*)&loaderGlobs.current);
+
+	//Check if we're in the gamedata section and copy over
+	if ((secCntr == LOADER_MAXSECTIONS) && (localLoaderSec != nullptr)) {
+		uint32* allocPtr = (undefined4*)Gods98::Mem_Alloc(strlen(section));
+		localLoaderSec->name = (char*)allocPtr;
+		strcpy((char*)allocPtr, section);
+
+		localLoaderSec->currentSize = 0;
+		localLoaderSec->totalSize = 0;
+		loaderGlobs.current = localLoaderSec;
+	}
+
+	//Setup flags and create callback
+	loaderGlobs.flags = loaderGlobs.flags | LOADER_GLOB_FLAG_ENABLED;
+	loaderGlobs.progressLast = -1.0;
+	Loader_FileLoadCallback(nullptr, 0, nullptr);
+	return;
+}
 
 // <LegoRR.exe @0044e180>
-//void __cdecl LegoRR::Loader_FileLoadCallback(const char* filename, uint32 fileSize, void* data);
+void __cdecl LegoRR::Loader_FileLoadCallback(const char* filename, uint32 fileSize, void* data)
+{
+	int x,y;
+	long long lVar2;
+	Size2F barSize;
+	Point2F barPosition;
 
-// <LegoRR.exe @0044e360>
-//void __cdecl LegoRR::Loader_display_shutdown(void);
+	float localFloat = fileSize;
+
+	if (((byte)loaderGlobs.flags & LOADER_GLOB_FLAG_ENABLED) != 0) {
+		if (((loaderGlobs.current == nullptr) || ((loaderGlobs.current)->currentSize = (loaderGlobs.current)->currentSize + localFloat, loaderGlobs.current == nullptr))
+			|| ((loaderGlobs.current)->totalSize == 0)) {
+			localFloat = 0.0f;
+		}
+		else {
+			//Get percentage here
+			float num = loaderGlobs.current->currentSize;
+			float deom = loaderGlobs.current->totalSize;
+			localFloat = num/deom;
+		}
```

Comment:
Avoid assignments inside if statements if they're not used for comparison.

In this case, Ghidra merged two separate if statement blocks to keep the `localFloat = 0.0f` assignment in one block. Control flow should be rewritten:
```cpp
if (loaderGlobs.current == nullptr) {
	localFloat = 0.0f;
}
else {
	loaderGlobs.current->currentSize += localFloat;
	if (loaderGlobs.current->totalSize == 0) {
		localFloat = 0.0f;
	}
	else {
		//Get percentage here
		float num = loaderGlobs.current->currentSize;
		float deom = loaderGlobs.current->totalSize;
		localFloat = num/deom;
	}
}
```